### PR TITLE
feat(type): add isNegativeZero

### LINKED ISSUE #784: isNegativeZero
Object.is can be used to provide the check under hood

[This PR](https://github.com/char0n/ramda-adjunct/pull/1067) should serve as a model template of PR for `isNegativeZero` function.

### PR DESCRIPTION
**Pull request by miguelcarvalho13**
Closes #784

### CODE REVIEW

File: src/isNegativeZero.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { identical } from 'ramda';
+
+/**
+ * Checks if value is a negative zero (-0).
+ *
+ * @func isNegativeZero
+ * @memberof RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/2.21.0|v2.21.0}
```
Comment (char0n) [inline src/isNegativeZero.js]: `@since {@link https://char0n.github.io/ramda-adjunct/2.22.0|v2.22.0}`

File: types/index.d.ts [hunk 2]
```diff
@@ -278,6 +278,11 @@ declare namespace RamdaAdjunct {
          */
         isNegative(val: any): val is number;
 
+        /**
+         * Checks if value is a negative zero (-0).
+         */
+        isNegativeZero(val: any): val is number;
```
Comment (char0n) [inline types/index.d.ts]: IMHO this is incorrect. 
```ts
isNegativeZero(val: any): boolean;
```
..would be more appropriate. This function can return `false` for number values.

File: src/isNegativeZero.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ * @category Type
+ * @sig * -> Boolean
+ * @param {*} val The value to test
+ * @return {boolean}
+ * @see {@link https://ramdajs.com/docs/#identical|R.identical}
```
Comment (char0n) [inline src/isNegativeZero.js]: We already have a PR for positive zero. Would be good to link to it.
`@see {@link https://ramdajs.com/docs/#identical|R.identical}`

I'd drop identical here, it's more an implementation detail that something that can really help you navigating the library.

File: src/isNegativeZero.js (hunk 1 above)
Comment (miguelcarvalho13) [reply src/isNegativeZero.js]: Done ✅

File: types/index.d.ts (hunk 2 above)
Comment (miguelcarvalho13) [reply types/index.d.ts]: Totally agree. Tbh I'm not that well familiar with Typescript but seeing that now it totally makes sense!

Done ✅

File: src/isNegativeZero.js (hunk 3 above)
Comment (miguelcarvalho13) [reply src/isNegativeZero.js]: Agreed!
Done ✅

**Comment by miguelcarvalho13** [comment]
> Very nice work! You uncovered very important fact to me and that is R.identical is using `Object.is` polyfill under the hood. Thanks. Pls address couple of comments that I had we can go for a merge.

I saw your suggestion to use `Object.is` and just to be sure I thought to search for that in the Ramda website and then I found that in the docs.

Btw, I'm very impressed with this `ramda-adjunct` repo its very well maintained and organized 👏

**Comment by char0n** [comment]
> I saw your suggestion to use Object.is and just to be sure I thought to search for that in the Ramda website and then I found that in the docs.

Yep, nice reasoning here, we avoided unnecessary polyfill.

> Btw, I'm very impressed with this ramda-adjunct repo its very well maintained and organized 

Thank you! With addition of your code it's going be even better! ;]

**Comment by char0n** [comment]
Ok I just resolved the code conflicts with master, let's wait for tests and we can go for a merge then. Thank you for contributing!

**Pull request merged**